@xen-orchestra/rest-api/src/pools/pool.controller.mts

```diff
@@ -0,0 +1,49 @@
+import { Example, Get, Path, Query, Response, Request, Route, Security, Tags } from 'tsoa'
+import { inject } from 'inversify'
+import { provide } from 'inversify-binding-decorators'
+import { Request as ExRequest } from 'express'
+import { RestApi } from '../rest-api/rest-api.mjs'
+import { notFoundResp, unauthorizedResp, type Unbrand } from '../open-api/common/response.common.mjs'
+import type { WithHref } from '../helpers/helper.type.mjs'
+import { XapiXoController } from '../abstract-classes/xapi-xo-controller.mjs'
+import type { XoPool } from '@vates/types'
+import { partialPools, pool, poolIds } from '../open-api/oa-examples/pool.oa-example.mjs'
+
+@Route('pools')
+@Security('*')
+@Response(unauthorizedResp.status, unauthorizedResp.description)
+@Tags('pools')
+@provide(PoolController)
+export class PoolController extends XapiXoController<XoPool> {
+  constructor(@inject(RestApi) restApi: RestApi) {
+    super('pool', restApi)
+  }
+
+  /**
+   *
+   * @example fields "name_label,id"
+   * @example filter "auto_poweron?"
```

Comment:
Try to always display the field you are filtering.
Don't forget to update your `partialPools`
```suggestion
   * @example fields "auto_poweron,name_label,id"
   * @example filter "auto_poweron?"
```